server/app/src/main/java/com/google/heartrate/wearos/app/gatt/attributes/GattCharacteristic.java

```diff
@@ -0,0 +1,240 @@
+package com.google.heartrate.wearos.app.gatt.attributes;
+
+import android.bluetooth.BluetoothDevice;
+import android.bluetooth.BluetoothGatt;
+import android.bluetooth.BluetoothGattCharacteristic;
+
+import com.google.heartrate.wearos.app.gatt.GattException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * {@link GattCharacteristic} is the wrapper over {@link BluetoothGattCharacteristic} class.
+ * <p>
+ * {@link GattCharacteristic} provides method which are common for all GATT characteristics.
+ * This abstraction makes all operations with {@link BluetoothGattCharacteristic}
+ * expandable for each specific characteristic type.
+ */
+public class GattCharacteristic implements GattAttribute {
+
+    /** {@link BluetoothGattCharacteristic} for characteristic. */
+    private final BluetoothGattCharacteristic mBluetoothGattCharacteristic;
+
+    /** Descriptors which characteristic include. */
+    private final HashMap<UUID, GattDescriptor> descriptorByUuid = new HashMap<>();
+
+    /**
+     * Configure {@link BluetoothGattCharacteristic} with given parameters.
+     * @param uuid characteristic uuid
+     * @param properties characteristic properties
+     * @param permissions characteristic permissions
+     */
+    public GattCharacteristic(UUID uuid, int properties, int permissions) {
+        mBluetoothGattCharacteristic =
+                new BluetoothGattCharacteristic(uuid, properties, permissions);
+    }
+
+    /**
+     * Configure {@link BluetoothGattCharacteristic} with given parameters and descriptors.
+     *
+     * @param uuid characteristic uuid
+     * @param properties characteristic properties
+     * @param permissions characteristic permissions
+     * @param gattDescriptors characteristic descriptors
+     */
+    public GattCharacteristic(UUID uuid, int properties, int permissions, GattDescriptor[] gattDescriptors) {
+        this(uuid, properties, permissions);
+
+        for (GattDescriptor gattDescriptor : gattDescriptors) {
+            descriptorByUuid.put(gattDescriptor.getUUid(), gattDescriptor);
+            mBluetoothGattCharacteristic.addDescriptor(gattDescriptor.getBluetoothGattDescriptor());
+        }
+    }
+
+    /**
+     * Determine weather characteristic has descriptor with given uuid or not.
```

Comment:
typo: whether / if

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/1/commits/f81955e6ee20af370c211fd96123abd538b599e1l)